lib/bowling_game.rb

```diff
@@ -0,0 +1,112 @@
+require_relative 'scorecard'
+
+class BowlingGame
+  attr_reader :current_frame, :strike, :spare, :scorecard
+
+  def initialize
+    @current_frame = 1
+    @strike = false
+    @spare = false
+    @scorecard = Scorecard.new
+  end
+
+  def roll_1(players_score)
+    if @strike
+      update_bonus(players_score)
+    elsif @spare
+      update_bonus(players_score)
+    end
+
+    if players_score < 10 && @current_frame == 10
+      update_roll_1_score(players_score)
+      update_roll_2_score(0)
+      update_roll_3_score(0)
+      update_scorecard
+    elsif players_score == 10 && @current_frame < 10
+      update_roll_1_score(players_score)
+      update_roll_2_score(0)
+      update_scorecard
+      @strike = true
+      "Strike!"
+    else
+      update_roll_1_score(players_score)
+      "Nice roll! Let's Roll again!"
+    end
+  end
```

Comment:
I feel like this is a bit unwieldy, I think there are elements that could be extracted into helper methods easily, especially where there are reused.

```suggestion
  def roll_1(players_score)
    update_bonus(players_score)  # extract all the logic from here to a helper method

    if players_score < 10 && @current_frame == 10 # extract the current_ftame == 10 to some method like tenth_frame?
      update_roll_1_score(players_score)
      update_roll_2_score(0)
      update_roll_3_score(0)
      update_scorecard
    elsif players_score == 10 && @current_frame < 10 # then use !tenth_frame?
      update_roll_1_score(players_score)
      update_roll_2_score(0)
      update_scorecard
      @strike = true
      "Strike!"
    else
      update_roll_1_score(players_score)
      "Nice roll! Let's Roll again!"
    end
  end
```